src/components/OrderConfirmation/OrderConfirmation.js

```diff
@@ -0,0 +1,157 @@
+
+import { Metadata } from "../Metadata";
+import { ActionStateIcon, ArrowRightIcon } from "../Icons";
+import { Panel } from "../Panel";
+import HighlightedLink from "../HighlightedLink";
+
+import { usePlans } from "../../contexts/plans";
+import { usePortalSettings } from "../../contexts/portal-settings";
+import accountsService from "../../services/accountsService";
+import { DATE_FORMAT } from "../../lib/config";
+
+import { PriceInfo } from "./PriceInfo";
+
+const State = {
+  Waiting: "Waiting",
+  Success: "Success",
+  Failure: "Failure",
+};
+
+const TITLE_BY_STATE = {
+  Waiting: "Processing payment",
+  Success: "Success!",
+  Failure: "Error occurred",
+};
+
+export const OrderConfirmation = ({ sessionId }) => {
+  const { settings } = usePortalSettings();
+  const { getPlanById } = usePlans();
+  const [state, setState] = React.useState(State.Waiting);
+  const [subscription, setSubscription] = React.useState(null);
+  const [isTakingTooLong, setIsTakingTooLong] = React.useState(false);
+
+  React.useEffect(() => {
+    const fetchSession = async () => {
+      try {
+        const sub = await accountsService
+          .get(`stripe/checkout/${sessionId}`, {
+            timeout: 30_000,
+            retry: 5,
+          })
+          .json();
+        setSubscription(sub);
+        setState(sub.status === "active" ? State.Success : State.Failure);
+      } catch (err) {
+        setState(State.Failure);
+        setSubscription(null);
+        console.log(err);
+      }
+    };
+
+    if (sessionId) {
+      fetchSession();
+
+      // Show a "This is taking longer than usual" message after 10s of waiting.
+      const timer = setTimeout(() => setIsTakingTooLong(true), 10_000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [sessionId, getPlanById]);
+
+  return (
+    <Panel className="px-3">
+      <Metadata>
+        <title>{TITLE_BY_STATE[state]}</title>
+      </Metadata>
+      <div className="w-full mb-8 flex flex-col gap-4 px-12 items-center">
+        <ActionStateIcon
+          size={100}
+          state={state}
+          className="transition-colors duration-300 ease-in-out"
+          waitingClass="!text-palette-200/70"
+          successClass="!text-primary"
+          failureClass="!text-error"
+        />
+        {state === State.Waiting && (
+          <div className="text-center">
+            <h4>Confirming your purchase...</h4>
+
+            {isTakingTooLong ? (
+              <p className="text-palette-400">Sorry, this is taking longer than usual.</p>
+            ) : (
+              <p className="font-bold">Please wait...</p>
+            )}
+          </div>
+        )}
+        {state === State.Failure && (
+          <div className="text-center">
+            <p className="font-bold mb-4">We encountered a problem contacting our payments provider.</p>
+            <p>
+              If you purchased a subscription, but don't see it on your account, please contact us at{" "}
```

Comment:
Something like this (the text is suggestion) would make the message less grievous. We could also replace the red error icon with orange warning triangle or sth like that.

> Your subscription status should be updated soon, please check back in couple of minutes and refresh the page, otherwise contact us at ...

Reply:
Good idea 🎯

Reply:
It does look better:
![image](https://user-images.githubusercontent.com/3853033/181738369-ff6120cb-bf0e-48d1-836c-191078dee368.png)

d642f78